samples/snippets/sample_preemptible.py

```diff
@@ -0,0 +1,148 @@
+    Returns:
+        Instance object.
+    """
+    instance_client = compute_v1.InstancesClient()
+    operation_client = compute_v1.ZoneOperationsClient()
+
+    # Describe the size and source image of the boot disk to attach to the instance.
+    disk = compute_v1.AttachedDisk()
+    initialize_params = compute_v1.AttachedDiskInitializeParams()
+    initialize_params.source_image = (
+        "projects/debian-cloud/global/images/family/debian-10"
+    )
+    initialize_params.disk_size_gb = 10
+    disk.initialize_params = initialize_params
+    disk.auto_delete = True
+    disk.boot = True
+    disk.type_ = "PERSISTENT"
+
+    # Use the default VPC network.
+    network_interface = compute_v1.NetworkInterface()
+    network_interface.name = "default"
+
+    # Collect information into the Instance object.
+    instance = compute_v1.Instance()
+    instance.name = instance_name
+    instance.disks = [disk]
+    instance.machine_type = f"zones/{zone}/machineTypes/e2-small"
+    instance.network_interfaces = [network_interface]
+
+    # Set the preemptible setting
+    instance.scheduling = compute_v1.Scheduling()
+    instance.scheduling.preemptible = True
+
+    # Prepare the request to insert an instance.
+    request = compute_v1.InsertInstanceRequest()
+    request.zone = zone
+    request.project = project_id
+    request.instance_resource = instance
+
+    # Wait for the create operation to complete.
+    print(f"Creating the {instance_name} instance in {zone}...")
+    operation = instance_client.insert_unary(request=request)
+    while operation.status != compute_v1.Operation.Status.DONE:
+        operation = operation_client.wait(
+            operation=operation.name, zone=zone, project=project_id
+        )
+    if operation.error:
+        print("Error during creation:", operation.error, file=sys.stderr)
+    if operation.warnings:
+        print("Warning during creation:", operation.warnings, file=sys.stderr)
+    print(f"Instance {instance_name} created.")
+    return instance_client.get(project=project_id, zone=zone, instance=instance_name)
+
+
+# [END compute_preemptible_create]
+
+
+# [START compute_preemptible_check]
+def is_preemptible(project_id: str, zone: str, instance_name: str) -> bool:
+    """
+    Check if a given instance is preemptible or not.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone you want to use. For example: “us-west3-b”
+        instance_name: name of the virtual machine to check.
+    Returns:
+        The preemptible status of the instance.
+    """
+    instance_client = compute_v1.InstancesClient()
+    instance = instance_client.get(
+        project=project_id, zone=zone, instance=instance_name
+    )
+    return instance.scheduling.preemptible
+
+
+# [END compute_preemptible_check]
+
+
+# [START compute_preemptible_history]
+def preemption_history(project_id: str, zone: str, instance_name: str = None):
```

Comment:
This function doesn't seem to have a corresponding test.

Reply:
It doesn't as there would always be an empty result. It's highly unlikely for an instance to be preempted in the time we have to run the test. This is not something we have control over, so the test would just be non-deterministic.

Reply:
Huh I see, thank you for the explanation. Can a test call this function (even if it's hard to assert anything about the output)? More in the spirit of making sure the request `op_client.list(req)` isn't malformed.

Reply:
OK, I rewrote this part a bit, to make the operation listing a separate function that can be used in testing :)